src/types.rs

```diff
@@ -1,5 +1,65 @@
 use std::collections::HashMap;
 
+use std::fmt;
+use std::cmp;
+use serde::*;
+use serde::de::{Visitor, Error, Unexpected};
+
+#[derive(Clone, Debug)]
+pub struct Timestamp {
+    string_repr: String,
+    float_repr: f64,
+}
+
+impl cmp::PartialEq for Timestamp {
```

Comment:
I think if we settle on a single value representation of `Timestamp` these can be `Derive`d.

Reply:
My current source of confusion on the Timestamp representation is this:

I want to use the timestamps for ordering, to insert some sort of unread marker, and handle scrollback and edits. So to me, that just means I want a thing that's ordered. But some API calls to Slack use the timestamp as a message ID; so I suspect things go haywire if you don't send them the same bits you got. Maybe we should be storing as a String then converting on-the-fly as needed for comparisons?

Reply:
I see, so that's why reading the value is actually hidden for now, and only the ordering is defined.

I would say it's better to store it as an `f64` because those are always easier to work with than `String`. We could implement `Display` for `Timestamp` to get the `String` representation.

Also, why did you change from `f32` to `f64`?

Reply:
I switched to `f64` because I was trying to use `From` instead of `as`, but I see that elsewhere this crate uses `f32`, so we'll stick with that.

Are we interested in exposing the actual value of a Timestamp directly? I'm not sure if that's a good idea, but it seems to be the general strategy across this crate to simply expose whatever we can deserialize.